stdlib/public/core/CMakeLists.txt

```diff
@@ -246,34 +246,42 @@ else()
   # effort has been completed.
   #set(LINK_FLAGS
   #  -Wl,--whole-archive swiftRuntime -Wl,--no-whole-archive)
-  if("${SWIFT_PATH_TO_LIBICU_BUILD}" STREQUAL "")
-    list(APPEND swift_core_private_link_libraries
-      ${SWIFT_${SWIFT_PRIMARY_VARIANT_SDK}_${SWIFT_PRIMARY_VARIANT_ARCH}_ICU_UC}
-      ${SWIFT_${SWIFT_PRIMARY_VARIANT_SDK}_${SWIFT_PRIMARY_VARIANT_ARCH}_ICU_I18N})
-  else()
+  if(NOT "${SWIFT_PATH_TO_LIBICU_BUILD}" STREQUAL "")
     list(APPEND swift_core_private_link_libraries -licui18nswift -licuucswift -licudataswift)
   endif()
 endif()
 
-if(SWIFT_PRIMARY_VARIANT_SDK STREQUAL ANDROID)
-  # workaround for libatomic needing to be after object files for swiftCore.so
-  list(APPEND swift_core_private_link_libraries atomic)
-elseif(SWIFT_PRIMARY_VARIANT_SDK STREQUAL CYGWIN)
+function(swift_core_private_libraries sdk arch libraries_var_name)
+  set(private_link_libraries)
+  if(NOT sdk IN_LIST SWIFT_APPLE_PLATFORMS AND "${SWIFT_PATH_TO_LIBICU_BUILD}" STREQUAL "")
+    list(APPEND private_link_libraries ${SWIFT_${sdk}_${arch}_ICU_UC}
+      ${SWIFT_${sdk}_${arch}_ICU_I18N})
+  endif()
+
+  if(sdk STREQUAL ANDROID)
+    # workaround for libatomic needing to be after object files for swiftCore.so
+    list(APPEND private_link_libraries atomic)
```

Comment:
This library isn't being linked here by the Android CI currently, maybe it isn't needed?

Reply:
@CodaFi, you reviewed the pull when this was put in, #9869, appreciate a review on this whole pull too when you have the time.

@drexin, can these two libatomic dependencies be removed too, like in #30155?

Reply:
I‘m not sure about Android. Can you remove them and see if it still compiles?

Reply:
Yep, I removed the libatomic dependency here and in `_add_variant_link_flags` and everything still compiled on an Android host. I guess this dependency can be removed for Android and BSD too?